charts/stable/mosquitto/values.yaml

```diff
@@ -30,7 +30,7 @@ configmap:
     enabled: true
     data:
       mosquitto.conf: |
-        listener {{ .Values.service.main.ports.main.port }}
+        listener {{ .Values.service.main.ports.main.targetPort }}
```

Comment:
Leave this at `port` and remove `targetPort: 1883` from main service.

that way `targetPort` will always be insync with `port`